pkg/sql/parser/lexer.go

```diff
@@ -131,6 +135,16 @@ func (l *lexer) Lex(lval *sqlSymType) int {
 			case ALL:
 				lval.id = USER_ALL
 			}
+		case ON:
+			switch nextID {
+			case DELETE:
+				lval.id = ON_DELETE
```

Comment:
can you keep this as `ON`, or does it shift-reduce conflict?

Reply:
This ends up in conflict. From my testing, I always have to disambiguate the usage of ON or else I get a conflict.